pkg/http/core/kubernetes/deploy.go

```diff
@@ -65,24 +70,50 @@ func (d *deployManfest) Run() error {
 		if err != nil {
 			return err
 		}
+		application := d.dm.Moniker.App
 
-		name := u.GetName()
-
-		err = d.kc.AddSpinnakerAnnotations(u, d.dm.Moniker.App)
+		err = d.kc.AddSpinnakerAnnotations(u, application)
 		if err != nil {
 			return err
 		}
 
-		err = d.kc.AddSpinnakerLabels(u, d.dm.Moniker.App)
+		err = d.kc.AddSpinnakerLabels(u, application)
 		if err != nil {
 			return err
 		}
 
+		if d.kc.IsVersioned(u) {
+			kind := strings.ToLower(u.GetKind())
+			namespace := u.GetNamespace()
+			name := u.GetName()
+			lo := metav1.ListOptions{
+				LabelSelector:  kubernetes.LabelKubernetesName + "=" + application,
+				TimeoutSeconds: &listTimeout,
+			}
+
+			results, err := client.ListResourcesByKindAndNamespace(kind, namespace, lo)
+			if err != nil {
+				return err
+			}
+
+			currentVersion := d.kc.GetCurrentVersion(results, kind, name)
+			latestVersion := d.kc.IncrementVersion(currentVersion)
+
+			u.SetName(u.GetName() + "-" + latestVersion.Short)
+
+			err = d.kc.AddSpinnakerVersionAnnotations(u, application, latestVersion)
+			if err != nil {
+				return err
+			}
+		}
+
 		meta, err := client.ApplyWithNamespaceOverride(u, d.dm.NamespaceOverride)
 		if err != nil {
 			return err
 		}
 
+		name := u.GetName()
```

Comment:
This needs to be a reference to the original name before versioning so that we store the cluster correctly.

For example, if I pass in a Kubernetes secret with name `my-secret`, it would be versioned and renamed to `my-secret-vNNN` however its cluster would still be `secret my-secret`.